src/Juvix/DependencyInfo.hs

```diff
@@ -0,0 +1,44 @@
+module Juvix.DependencyInfo where
+
+import Data.Graph (Graph, Vertex)
+import Data.Graph qualified as Graph
+import Data.HashMap.Strict qualified as HashMap
+import Data.HashSet qualified as HashSet
+import Juvix.Prelude
+
+-- DependencyInfo is polymorphic to anticipate future use with other identifier
+-- types in JuvixCore and further. The graph algorithms don't depend on the
+-- exact type of names (the polymorphic type n), so there is no reason to
+-- specialise DependencyInfo to any particular name type
+data DependencyInfo n = DependencyInfo
+  { _depInfoGraph :: Graph,
+    _depInfoNodeFromVertex :: Vertex -> (n, HashSet n),
+    _depInfoVertexFromName :: n -> Maybe Vertex,
+    _depInfoReachable :: HashSet n
+  }
+
+makeLenses ''DependencyInfo
+
+createDependencyInfo :: (Hashable n, Ord n) => HashMap n (HashSet n) -> HashSet n -> DependencyInfo n
+createDependencyInfo edges startNames =
+  DependencyInfo
+    { _depInfoGraph = graph,
+      _depInfoNodeFromVertex = \v -> case nodeFromVertex v of (_, x, y) -> (x, HashSet.fromList y),
+      _depInfoVertexFromName = vertexFromName,
+      _depInfoReachable = reachableNames
+    }
+  where
+    (graph, nodeFromVertex, vertexFromName) =
```

Comment:
add type signatures for `graph`, `nodeFromVertex`, `vertexFromName`, `reahableNames` and `startVertices`.
We usually prefer to have type signatures for everything in a where clause

Reply:
Done